src/core/Akka.Persistence/Snapshot/MemorySnapshotStore.cs

```diff
@@ -9,69 +9,73 @@
 using System.Collections.Generic;
 using System.Linq;
 using System.Threading.Tasks;
+using Akka.Util.Internal;
 
 namespace Akka.Persistence.Snapshot
 {
+    /// <summary>
+    /// INTERNAL API.
+    ///
+    /// In-memory SnapshotStore implementation.
+    /// </summary>
     public class MemorySnapshotStore : SnapshotStore
     {
-        private readonly List<SnapshotEntry> _snapshotCollection = new List<SnapshotEntry>();
+        private readonly object _ssLock = new object();
 
         /// <summary>
         /// This is available to expose/override the snapshots in derived snapshot stores
         /// </summary>
-        protected virtual List<SnapshotEntry> Snapshots { get { return _snapshotCollection; } }
+        protected virtual List<SnapshotEntry> Snapshots { get; } = new List<SnapshotEntry>();
 
         protected override Task DeleteAsync(SnapshotMetadata metadata)
         {
-            Func<SnapshotEntry, bool> pred = x => x.PersistenceId == metadata.PersistenceId &&
-            (metadata.SequenceNr <= 0 || metadata.SequenceNr == long.MaxValue || x.SequenceNr == metadata.SequenceNr) &&
-            (metadata.Timestamp == DateTime.MinValue || metadata.Timestamp == DateTime.MaxValue || x.Timestamp == metadata.Timestamp.Ticks);
+            bool Pred(SnapshotEntry x) => x.PersistenceId == metadata.PersistenceId && (metadata.SequenceNr <= 0 || metadata.SequenceNr == long.MaxValue || x.SequenceNr == metadata.SequenceNr)
+                                                                                    && (metadata.Timestamp == DateTime.MinValue || metadata.Timestamp == DateTime.MaxValue || x.Timestamp == metadata.Timestamp.Ticks);
+            
 
-            return Task.Run(() =>
```

Comment:
The root cause of the issues the `MemorySnapshotStore` had were that we're using a `List<T>`, which is not thread safe, and were kicking off all of our operations inside the `SnapshotStore` asynchronously via `Task.Run`. 

My solution was to just get rid of the `Task.Run` calls and have all of the work done to the in-memory collection occur synchronously inside the `SnapshotStore` actor. This caused the test to go from fail to pass.